pkg/interceptors/github/github.go

```diff
@@ -96,7 +129,306 @@ func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequ
 		}
 	}
 
+	// For event types pull_request, issue_comment check github owners approval is required
+	// User can specify both event type or any one of them
+	if p.GithubOwners.Enabled {
+		ownerCheckAllowed := false
+		for _, allowedEvent := range ownersEventTypes {
+			if actualEvent == allowedEvent {
+				ownerCheckAllowed = true
+				break
+			}
+		}
+		if ownerCheckAllowed {
```

Comment:
Can we do if !ownerCheckAllowed { return ...} to reduce the nesting a bit: https://danp.net/posts/reducing-go-nesting/